src/parse.c

```diff
@@ -3150,6 +3151,17 @@ gboolean
 netplan_parser_load_yaml(NetplanParser* npp, const char* filename, GError** error)
 {
     yaml_document_t *doc = &npp->doc;
+    /* Log a warning if a file can be read or written by a non-owner.
+     * It could contain sensitive information (e.g. WiFi passwords), so should
+     * stay secret. */
+    mode_t mask = S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
+    struct stat info;
+    if (stat(filename, &info) < 0) {
+        g_set_error(error, G_FILE_ERROR, errno, "Cannot stat %s: %s",
+                    filename, strerror(errno));
+        return FALSE;
+    } else if (info.st_mode & mask)
+        g_warning("Weak permissions on %s", filename);
```

Comment:
suggestion: we probably should tell the user what is the correct permission they must use. Something like `"Weak permissions on %s (recommended is 600)"`

Or something like openssh does: `"Permissions for %s are too open. Netplan files should NOT be accessible by others."`